From 2bc68477023015b2ab3973057e90d36fa6f93923 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Fri, 30 Mar 2018 14:01:01 +0200
Subject: [PATCH] cpio: fix regression with symlinks

Fix to make cpio create symlinks.

This fixes a regression in cpio introduced with
a84db18fc71d09e801df0ebca048d82e90b32c6a (tar,unzip: postpone creation
of symlinks with "suspicious" targets).

Signed-off-by: Natanael Copa <ncopa@alpinelinux.org>
---
 archival/cpio.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/archival/cpio.c b/archival/cpio.c
index 1d6cbd1e2..3eb37c603 100644
--- a/archival/cpio.c
+++ b/archival/cpio.c
@@ -507,6 +507,7 @@ int cpio_main(int argc UNUSED_PARAM, char **argv)
 	archive_handle->cpio__blocks = (off_t)-1;
 	while (get_header_cpio(archive_handle) == EXIT_SUCCESS)
 		continue;
+	create_symlinks_from_list(archive_handle->symlink_placeholders);
 
 	if (archive_handle->cpio__blocks != (off_t)-1
 	 && !(opt & OPT_QUIET)
-- 
2.16.3

